modules/repository/adopt.go

```diff
@@ -154,7 +154,7 @@ func ListUnadoptedRepositories(query string, opts *db.ListOptions) ([]string, in
 	count := 0
 
 	// We're going to iterate by pagesize.
-	root := filepath.Join(setting.RepoRootPath)
+	root := filepath.Clean(setting.RepoRootPath)
```

Comment:
Just to be safe - I'm keeping the `.Clean()` part in, as `.Walk()` will not transform root of any kind and just pass it into the syscall as it was given.

Reply:
But I think we should do that on setting read function but not here.

Reply:
Otherwise LGTM.

Reply:
> But I think we should do that on setting read function but not here.

That's fine by me - but that could theoretical break existing installs as they may contain non-cleaned `RepoRootPath` setting.